src/voter/voting_round.rs

```diff
@@ -273,7 +289,13 @@ impl<H, N, E: Environment<H, N>> VotingRound<H, N, E> where
 			}
 		});
 
-		prevotes.chain(precommits).collect()
+		let prevotes_len = prevotes.len();
+
+		HistoricalVotes {
+			seen: prevotes.chain(precommits).collect(),
+			prevote_idx: self.votes.prevote_idx(),
+			precommit_idx: self.votes.precommit_idx().map(|idx| idx + prevotes_len),
```

Comment:
I don't think this logic is right. There is no guarantee that we see all prevotes before seeing any precommits. Since the estimate logic depends on both prevotes and precommits, we need an _interleaved_ list where the indices are those we've voted at.

Reply:
Ok, I thought it doesn't invalidate the votes.